haas/model.py

```diff
@@ -227,6 +182,31 @@ def get_console(self):
     def get_console_log_filename(self):
         return '/var/run/haas_console_logs/%s.log' % self.ipmi_host
 
+class Obm(AnonModel):
+    """ Obm superclass supporting various drivers 
+        related to out of band management of servers """
+    type = Column(String, nullable=False)
+
+    __mapper_args__ = {
+            'polymorphic_identity': 'new_obmType',
+            'polymorphic_on': type
+            }
+    @staticmethod
+    def validate(kwargs):
+        """
+        Verify that ``kwargs`` is a legal set of keywords args to ``__init__``
+        Raise a ``schema.SchemaError`` if the  ``kwargs`` is invalid. 
+        Note well: This is a *static* method; it will be invoked on the class.
+        """
+        assert False, "Subclasses MUST override the validate method "
+
+    def power_cycle(self):
+        """
+        Depending on which sub-class is associated with a given node,
+        The exact method of power cycle will differ Implements the power cycle method specific to 
+        each 'TYPE' of obm pertaining to the respective node
```

Comment:
You don't really need to explain the whole inheritance concept here; just explain what the power_cycle method is supposed to do. Just mention in the class's docstring the fact that it's meant to be subclassesd, and which methods need to be overwritten (validate and power_cycle). Have a look at `Switch` for a reference.